boxsdk/object/retention_policy.py

```diff
@@ -0,0 +1,88 @@
+# coding: utf-8
+from __future__ import unicode_literals, absolute_import
+import json
+
+from .base_object import BaseObject
+from ..pagination.marker_based_object_collection import MarkerBasedObjectCollection
+
+
+class RetentionPolicy(BaseObject):
+    """Represents a Box retention policy."""
+    _item_type = 'retention_policy'
+
+    def get_url(self, *args):
+        """
+        Returns the url for this retention policy.
+        """
+        return self._session.get_url('retention_policies', self._object_id, *args)
+
+    def assign(self, assignee, fields=None):
+        """Assign a retention policy to a Box item
+
+        :param assignee:
+            The item to assign the retention policy on.
+        :type assignee:
+            :class:`Folder`, :class:`Enterprise`, or :class:`MetadataTemplate`
+        :param fields:
+            List of fields to request.
+        :type fields:
+            `Iterable` of `unicode`
+        :returns:
+            A :class:`RetentionPolicyAssignment` object.
+        :rtype:
+            :class:`RetentionPolicyAssignment`
+        """
+        url = self._session.get_url('retention_policy_assignments')
+        body = {
+            'policy_id': self.object_id,
+            'assign_to': {
+                'type': assignee.object_type,
+                'id': assignee.object_id,
+            }
+        }
+        params = {}
+        if fields is not None:
+            params['fields'] = ','.join(fields)
+        response = self._session.post(url, data=json.dumps(body), params=params).json()
+        return self.translator.translate(response['type'])(
+            session=self._session,
+            object_id=response['id'],
+            response_object=response,
+        )
+
+    def assignments(self, assignment_type=None, limit=None, marker=None, fields=None):
+        """Get the assignments for the retention policy.
+
+        :param assignment_type:
+            The type of retention policy assignment to retrieve.
```

Comment:
Might be helpful to provide some examples of what goes here (e.g. `folder`, `enterprise`)